elastic/beats
# Filebeat input v2 s3

- Refactoring

## What does this PR do?

Move s3 input to input v2 API.

This change splits the internal s3Input into s3Input and s3Collector. The s3Input is responsible for configuration only.

The unit tests have been modified, but the integration tests need some more work. 

## Why is it important?

Update to v2 input API.

## Checklist



~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## Author's Checklist



## How to test this PR locally

Manual testing:
1. Create an SQS queue and an S3 bucket in the same AWS region using Amazon SQS console.
2. Configure SQS queue by replacing the access policy attached to the queue with the following queue policy:
```
{
 "Version": "2012-10-17",
 "Id": "example-ID",
 "Statement": [
  {
   "Sid": "example-statement-ID",
   "Effect": "Allow",
   "Principal": {
    "AWS":"*"  
   },
   "Action": [
    "SQS:SendMessage"
   ],
   "Resource": "<SQS-queue-ARN>",
   "Condition": {
      "ArnLike": { "aws:SourceArn": "arn:aws:s3:*:*:<bucket-name>" }
   }
  }
 ]
}
```
3. Using the Amazon S3 console, add a notification configuration requesting Amazon S3 to publish events of the `s3:ObjectCreated:*` type to your Amazon SQS queue.
4. Upload an object to the S3 bucket and verify the event notification in the Amazon SQS console.

For running the `s3_integration_test.go`, environment variables are needed. For example:
```
QUEUE_URL=https://sqs.us-east-1.amazonaws.com/428152502467/test-fb-ks;AWS_PROFILE_NAME=elastic-beats;S3_BUCKET_REGION=us-east-1;S3_BUCKET_NAME=test-fb-ks
```

## Related issues


-
